esphome/components/bl0910/bl0910.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/components/spi/spi.h"
+
+#include "esphome/components/sensor/sensor.h"
+
+namespace esphome {
+namespace bl0910 {
+
+// Mode 1 Trailing clock CPHA=1, Polarity low CPOL=0
+
+#define NUM_CHANNELS 11
+
+class BL0910 : public PollingComponent,
+               public spi::SPIDevice<spi::BIT_ORDER_MSB_FIRST, spi::CLOCK_POLARITY_LOW, spi::CLOCK_PHASE_TRAILING,
+                                     spi::DATA_RATE_1MHZ> {
+ public:
+  void set_voltage_sensor(sensor::Sensor *voltage_sensor_, int index, float uref_) {
+    voltage_sensor[index - 1] = voltage_sensor_;
+    uref[index - 1] = uref_;
+  }
+  void set_current_sensor(sensor::Sensor *current_sensor_, int index, float iref_) {
+    current_sensor[index - 1] = current_sensor_;
+    iref[index - 1] = iref_;
+  }
+  void set_power_sensor(sensor::Sensor *power_sensor_, int index, float pref_) {
+    power_sensor[index - 1] = power_sensor_;
+    pref[index - 1] = pref_;
+  }
+  void set_energy_sensor(sensor::Sensor *energy_sensor_, int index, float eref_) {
+    energy_sensor[index - 1] = energy_sensor_;
+    eref[index - 1] = eref_;
+  }
+  void set_power_factor_sensor(sensor::Sensor *power_factor_sensor_, int index) {
+    power_factor_sensor[index - 1] = power_factor_sensor_;
+  }
+  void set_frequency_sensor(sensor::Sensor *frequency_sensor_) { frequency_sensor = frequency_sensor_; }
+  void set_temperature_sensor(sensor::Sensor *temperature_sensor_) { temperature_sensor = temperature_sensor_; }
+
+  void loop() override;
+
+  void update() override;
+  void setup() override;
+  void dump_config() override;
+
+ protected:
+  sensor::Sensor *voltage_sensor[NUM_CHANNELS] = {};
```

Comment:
lower_snake_case_with_trailing_underscore_
```suggestion
  sensor::Sensor *voltage_sensor_[NUM_CHANNELS] = {};
```